agents/hwp_rotation/rotation_agent.py

```diff
@@ -0,0 +1,244 @@
+
+            self.switching = True
+            time.sleep(1)
+            self.cmd.user_input('on')
+            self.switching = False
+
+        return True, 'Set Kikusui on'
+
+    def set_off(self, session, params = None):
+        with self.lock.acquire_timeout(0, job = 'set_off') as acquired:
+            if not acquired:
+                self.log.warn('Could not set off because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            self.switching = True
+            time.sleep(1)
+            self.cmd.user_input('off')
+            self.switching = False
+
+        return True, 'Set Kikusui off'
+
+    def set_v(self, session, params = None):
+        if params == None:
+            params = {'volt': 0}
+
+        with self.lock.acquire_timeout(0, job = 'set_v') as acquired:
+            if not acquired:
+                self.log.warn('Could not set v because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            self.switching = True
+            time.sleep(1)
+            self.cmd.user_input('V {}'.format(params['volt']))
+            self.switching = False
+
+        return True, 'Set Kikusui voltage to {} V'.format(params['volt'])
+
+    def use_ext(self, session, params = None): 
+        with self.lock.acquire_timeout(0, job = 'use_ext') as acquired:
+            if not acquired:
+                self.log.warn('Could not use external voltage because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            self.switching = True
+            time.sleep(1)
+            self.cmd.user_input('U')
+            self.switching = False
+
+        return True, 'Set Kikusui voltage to PID control'
+
+    def ign_ext(self, session, params = None):
+        with self.lock.acquire_timeout(0, job = 'ign_ext') as acquired:
+            if not acquired:
+                self.log.warn('Could not ignore external voltage because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            self.switching = True
+            time.sleep(1)
+            self.cmd.user_input('I')
+            self.switching = False
+
+        return True, 'Set Kikusui voltage to direct control'
+
+    def start_IV_acq(self, session, params = None):
+        with self.lock.acquire_timeout(timeout = 0, job = 'IV_acq') as acquired:
+            if not acquired:
+                self.log.warn('Could not start IV acq because {} is already running'
+                              .format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            session.set_status('running')
+        
+        self.take_data = True
+
+        while self.take_data:
+            data = {'timestamp': time.time(), 'block_name': 'HWPKikusui_IV', 'data': {}}
+            
+            if not self.switching:
```

Comment:
I tried removing this attribute but ran into issues. It's purpose is to prevent the process from sending a read command while some other command is already being run

Reply:
Ah, so you're trying to leave the process running while running separate tasks. That scenario with the `TimeoutLock` is outlined here: https://ocs.readthedocs.io/en/develop/developer/agents.html#release-and-reacquire

Reply:
I believe the issue here was the main iv_acq process loop was not within the timeout lock's context, so could be executed simultaneously with any of the tasks that also communicate with the power supply. Fixing this in the work I'm doing on this. Marking as resolved here.